node/pegnet/addresses.go

```diff
@@ -183,6 +183,53 @@ func (p *Pegnet) SelectBalance(adr *factom.FAAddress, ticker fat2.PTicker) (uint
 	return balance, nil
 }
 
+type BalancePair struct {
+	Address *factom.FAAddress
+	Balance uint64
+}
+type BalancesPair struct {
+	Address  *factom.FAAddress
+	Balances []uint64
+}
+
+// SelectRichList returns the balance of all addresses for a given ticker
+func (p *Pegnet) SelectRichList(ticker fat2.PTicker, count int) ([]BalancePair, error) {
+	if ticker <= fat2.PTickerInvalid || fat2.PTickerMax <= ticker {
+		return nil, fmt.Errorf("invalid token type")
+	}
+	if count < 1 {
+		return nil, fmt.Errorf("invalid count")
+	}
+	var res []BalancePair
+	stmtStringFmt := `SELECT address, %[1]s_balance FROM pn_addresses WHERE %[1]s_balance > 0 ORDER BY %[1]s_balance DESC LIMIT ?;`
+	stmt := fmt.Sprintf(stmtStringFmt, strings.ToLower(ticker.String()))
+	rows, err := p.DB.Query(stmt, count)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
```

Comment:
Should we return an empty array vs a nil?

Reply:
nope, returning nil here is fine. i think it only matters when we convert to json that we need an empty uninitialized array vs nil pointer. golang should treat a nil pointer array the same as an empty array